api/v1alpha1/api_webhook.go

```diff
@@ -151,41 +151,42 @@ func (a *APIValidator) InjectDecoder(d *admission.Decoder) error {
 	return nil
 }
 
-func (a *APIValidator) PathAlreadyDeployed(ctx context.Context, fleet *EnvoyFleetID, newApi API) error {
-	var apiObjs APIList
+func (a *APIValidator) PathAlreadyDeployed(ctx context.Context, newApi API) error {
+	apiObjs := &APIList{}
 	parser := spec.NewParser(nil)
 	newApiSpec, err := parser.ParseFromReader(strings.NewReader(newApi.Spec.Spec))
 	if err != nil {
 		return err
 	}
 
 	// Get all API objects with this fleet field set
-	if err := a.Client.List(ctx, &apiObjs, &client.ListOptions{}); err != nil {
+	if err := a.Client.List(ctx, apiObjs, &client.ListOptions{}); err != nil {
 		return fmt.Errorf("failure querying for the deployed APIs: %w", err)
 	}
-
 	duplicates := openapi3.Paths{}
 	// filter out apis are in the process of deletion
 	for _, api := range apiObjs.Items {
-		if (api.Name != newApi.Name && api.Namespace != newApi.Namespace) &&
-			api.Spec.Fleet.Name == fleet.Name &&
-			api.Spec.Fleet.Namespace == fleet.Namespace &&
-			api.ObjectMeta.DeletionTimestamp.IsZero() {
-			apiSpec, err := parser.ParseFromReader(strings.NewReader(api.Spec.Spec))
-			if err != nil {
-				return err
-			}
+		if api.Name == newApi.Name && api.Namespace == newApi.Namespace {
+			continue
+		}
 
-			for path := range newApiSpec.Paths {
-				if p, ok := apiSpec.Paths[path]; ok {
-					duplicates[path] = p
-				}
+		if api.Spec.Fleet.Name != newApi.Spec.Fleet.Name && api.Spec.Fleet.Namespace != newApi.Spec.Fleet.Namespace {
+			continue
+		}
+		apiSpec, err := parser.ParseFromReader(strings.NewReader(api.Spec.Spec))
+		if err != nil {
+			return err
+		}
+
+		for path := range newApiSpec.Paths {
+			if p, ok := apiSpec.Paths[path]; ok {
+				duplicates[path] = p
 			}
 		}
 	}
 
 	if len(duplicates) > 0 {
-		return fmt.Errorf("paths %s already exist with envoyfleet %s", reflect.ValueOf(duplicates).MapKeys(), fleet)
+		return fmt.Errorf("paths %s already exist with envoyfleet %s/%s", reflect.ValueOf(duplicates).MapKeys(), newApi.Spec.Fleet.Name, newApi.Spec.Fleet.Namespace)
```

Comment:
I'd probably add `newApi.Namespace` here as well.

Not a big deal.

Reply:
I already did so
```
... newApi.Spec.Fleet.Name, newApi.Spec.Fleet.Namespace)
```
see the end of the println line